### PR TITLE
Singleclick show hide label

### LINKED ISSUE #4020: [needs-docs][labeling] When using the Show/Hide Label tool, a single
click on a label toggles this label
Original commit: https://github.com/qgis/QGIS/commit/d8a3031d65417075d0aa2ab8c94b6d177836d2e6 by nyalldawson

Unlike in marquee mode, where users must hold ctrl while dragging
over labels to hide them, when only a click on an existing label
is made we trigger a toggle of the label. It's a more intuitive
UX.

### PR DESCRIPTION
**Pull request by lajunek**
Goal: Users shall be able to show/hide labels. This works without setting the data-defined options.

Ticket(s): fix #4020

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -1513,8 +1513,7 @@ The Label Toolbar
 
 The :guilabel:`Label Toolbar` provides some tools to manipulate |labeling|
 :ref:`label <vector_labels_tab>` or |diagram| :ref:`diagram <sec_diagram>`
-properties, but only if the corresponding data-defined option is indicated
-(otherwise, buttons are disabled).
+properties but only if the corresponding data-defined option is indicated.
```
Comment (havatv) [inline docs/user_manual/working_with_vector/vector_properties.rst:1516]:
```suggestion
properties, but only if the corresponding data-defined option is indicated.
```

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -1529,7 +1528,7 @@ same way with diagrams:
 
 * |pinLabels| :sup:`Pin/Unpin Labels And Diagrams` that has data-defined
   position. By clicking or draging an area, you pin label(s). If you click or
-  drag an area holding :kbd:`Shift`, label(s) are unpinned. Finally, you can
+  drag an area holding :kbd:`Shift` label(s) are unpinned. Finally, you can
```
Comment (havatv) [inline docs/user_manual/working_with_vector/vector_properties.rst:1531]:
```suggestion
  drag an area holding :kbd:`Shift`, label(s) are unpinned. Finally, you can
```

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -1539,12 +1538,14 @@ same way with diagrams:
   determine whether any important labels are missing from the maps (e.g. due
   to overlaps or other constraints). They are displayed with a customizable
   color (see :ref:`automated_placement`).
-* |showHideLabels| :sup:`Show/Hide Labels And Diagrams` that has
-  data-defined visbility. If you click or drag an area holding :kbd:`Shift`,
-  then label(s) are hidden. When a label is hidden, you just have to click
-  or drag an area around the feature's point to restore its visibility.
+* |showHideLabels| :sup:`Show/Hide Labels And Diagrams`. If you click on the labels,
+  then they are hidden. When a label is hidden, you just have to click
```
Comment (havatv) [inline docs/user_manual/working_with_vector/vector_properties.rst:1542]:
```suggestion
  or click and drag an area holding :kbd:`Shift`, they are hidden.
  When a label is hidden, you just have to click
```

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -1539,12 +1538,14 @@ same way with diagrams:
...
+  on the feature to restore its visibility. This tool also works without setting
```
Comment (havatv) [inline docs/user_manual/working_with_vector/vector_properties.rst:1543]:
```suggestion
  on the feature to restore its visibility.
  If you drag an area, all the labels in the area will be restored.
  This tool also works without setting
```

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 5]
```diff
@@ -1539,12 +1538,14 @@ same way with diagrams:
...
+  data-defined options.
 * |moveLabel| :sup:`Move Label And Diagram` that has data-defined
-  position. You just have to drag the label to the desired place.
+  position. You just have to drag the label to the desired place. This only
+  works when you set the data-defined override for the :guilabel:`Data defined`
+  entries.
```
Comment (havatv) [inline docs/user_manual/working_with_vector/vector_properties.rst:1548]: I am not sure if I understand this. Does "Move Label And Diagram" only work for data-defined positions?
I was able to use the tool to move labels that were "Single Labels", without having anything "data-defined".
Perhaps "data-defined" here means something else (that the location of labels is determined from feature positions, and not static?).

Comment (lajunek) [reply docs/user_manual/working_with_vector/vector_properties.rst:1549]: @hatav I checked the Label Toolbar: all tools work fine without setting the data-defined coordinates

**Comment by lajunek** [comment]
@havatv do we need to do something so that Travis works here?

**Comment by havatv** [comment]
> @havatv do we need to do something so that Travis works here?

Travis seems to still be busy testing. Let's wait until it has completed.

**Comment by havatv** [comment]
Travis did not report anything, and from the logs it seemed to have completed (even if it was not all "green"). So should be OK to merge.

**Pull request merged**